### PR TITLE
Docker Gradle Integration

### PR DESCRIPTION
**Pull request by bgulla**
Myriad can now build the ResourceManager docker container straight from gradle.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,6 +21,23 @@ allprojects {
...
+    dependencies {
+        classpath 'com.bmuschko:gradle-docker-plugin:2.6.1'
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+
```
Comment (DarinJ) [inline build.gradle]: nit: 3 new lines.

File: docker/Dockerfile [hunk 2]
```diff
@@ -43,8 +43,8 @@ ADD myriad-bin/install-yarn.sh /install-yarn.sh
 RUN sh /install-yarn.sh
 
 # Copy over myriad libraries
-ADD /libs/myriad-executor-runnable-0.1.0.jar /usr/local/libexec/mesos/
-ADD /libs/* /usr/local/hadoop/share/hadoop/yarn/lib/
+ADD ./libs/myriad-executor-runnable-*.jar /usr/local/libexec/mesos/
+ADD ./libs/* /usr/local/hadoop/share/hadoop/yarn/lib/
```
Comment (DarinJ) [inline docker/Dockerfile]: Different PR but myriad-executor-runnable is no longer needed we should remove.

File: docker/ResourceManager.dockerfile [hunk 3]
```diff
@@ -44,8 +43,9 @@ ADD myriad-bin/install-yarn.sh /install-yarn.sh
 RUN sh /install-yarn.sh
 
 # Copy over myriad libraries
-ADD /libs/myriad-executor-runnable-0.1.0.jar /usr/local/libexec/mesos/
-ADD /libs/* /usr/local/hadoop/share/hadoop/yarn/lib/
+ADD ./libs/myriad-executor-runnable-0.1.0.jar /usr/local/libexec/mesos/
+ADD ./libs/* /usr/local/hadoop/share/hadoop/yarn/lib/
+
```
Comment (DarinJ) [inline docker/ResourceManager.dockerfile]: Again how is myriad-executor-runnable-0.1.0.jar, I think it's deprecated.

File: docker/build.gradle [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    mavenCentral()
+}
+
+apply plugin: 'com.bmuschko.docker-remote-api'
+
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+
+task copyExecutorJar(type:Copy){
+    from '../myriad-executor/build/libs/myriad-executor-runnable-0.1.0.jar'
```
Comment (sarjeet2013) [inline docker/build.gradle]: As Darin suggested in other places, this seems deprecated and no need to copy this.

File: docker/build.gradle [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    into 'libs'
+}
+
+task copyMyriadSchedulerJars(type:Copy){
```
Comment (sarjeet2013) [inline docker/build.gradle]: Looks like a dup of below 'copySchedulerJars' method. This can be removed, it is not used anywhere.

**Comment by bgulla** [comment]
back from the grave. finally pushed changes.

**Comment by DarinJ** [comment]
LGTM going to merge tomorrow if no one else has any comments.

**Pull request closed without merging**